pkg/storage/intent_resolver.go

```diff
@@ -302,26 +265,6 @@ func (ir *intentResolver) maybePushTransactions(
 // differently and would be better served by different entry points,
 // but combining them simplifies the plumbing necessary in Replica.
 func (ir *intentResolver) processIntentsAsync(r *Replica, intents []intentsWithArg) {
```

Comment:
By the way, we're passing `wait=true` here to `RunLimitedAsyncTask` means that something which horks the intent queue will starve other work. I think you might consider what's being done `DistSender.sendPartialBatchAsync`. There, we pass `wait=false` to `Stopper.RunLimitedAsyncTask`, and on `stop.ErrThrottled`, we synchronously send the partial batch.